smart-turret/packages/client/src/App.tsx

```diff
@@ -1,21 +1,154 @@
+//Import packages
 import { useMUD } from "./MUDContext";
+import React, { useEffect, useState, useRef } from "react";
 
-const styleUnset = { all: "unset" } as const;
+//Import EVE Frontier Packages
+import { EveButton, EveInput, EveScroll } from "@eveworld/ui-components";
+
+//Import CSS
+import "./styles.css";
+import "./styles-ui.css";
+
+//Import components
+import WhitelistEntry from './components/WhitelistEntry'
+import AppContainer from './components/AppContainer'
+import Title from './components/Title'
+import ContentContainer from './components/ContentContainer'
+import Section from './components/Section'
 
 export const App = () => {
-  const {
-    network: { tables, useStore }
-  } = useMUD();
-
-  const tasks = useStore((state) => {
-    const records = Object.values(state.getRecords(tables.Tasks));
-    records.sort((a, b) => Number(a.value.createdAt - b.value.createdAt));
-    return records;
-  });
-
-  return (
-    <>
-
-    </>
-  );
-};
+	const {
+		systemCalls: {
+			getWhitelist,
+			addToWhitelist,
+			removeFromWhitelist
+		},
+	} = useMUD();
+
+	//Character name input value
+	const characterNameInputRef = useRef(0);
+
+	//Pre-populate the UI list 
+	const [whitelist, setWhitelist] = useState(<WhitelistEntry id={"LOADING...."}></WhitelistEntry>);
+	const [addErrorVar, setAddErrorVar] = useState("");
+
+	//Edit character name
+	const handleEdit = (
+		refString: React.MutableRefObject<number>,
+		eventString: number
+	): void => {
+		refString.current = eventString;
+	};
+
+	//Initial load
+	useEffect(() => {
+		setTimeout(function(){
+			fetchWhitelist(null);
+		}, 1000)
+	}, [])
+
+	//Remove from the whitelist
+	const remove = async (id) => {
+		const whitelist = await removeFromWhitelist(id);
+
+		loadWhitelist(whitelist);
+	}
+
+	//Fetch the whitelist data
+	async function fetchWhitelist(event){
+		if(event) event.preventDefault();		
+		const whitelist = await getWhitelist();
+		loadWhitelist(whitelist);
+	}	
+
+	//Load the whitelist UI
+	async function loadWhitelist(result){
+		if(result == null) return;
+		
+		var newArray = result.whitelist.map((value) => <WhitelistEntry key={value.toString()} id={value.toString()} handleClick={remove}>{value}</WhitelistEntry>)
+		setWhitelist(newArray);
+	}
+
+	//Add to the whitelist
+	async function addToWhitelistButton (){
+		//Check there is a name
+		if(characterNameInputRef.current == ""){			
+			setAddErrorVar("- NO INPUT");
+			return;
+		}
+
+		//Fetch the world API data
+		const response = await fetch(`https://blockchain-gateway-nova.nursery.reitnorf.com/smartcharacters`);
+		const data = await response.json();			
+
+		var address = null;
+
+		//Check to see if the address exists
+		for(var i = 0; i < data.length; i++){
+			if(data[i].name == characterNameInputRef.current){
```

Comment:
Could you go by characterId address instead of name ? Or possibly enter characterName or characterId or characterAddress